Mostly tech level changes

fluffyfreak opened the pull request:
<!-- Please describe new feature, possible with screenshot if needed. -->
This PR bumps the tech level for stations based on both population and whether or not it's a homeworld for a faction.

More could be done here, taking into account the distance from the homeworld divided by the radius of the factions bubble for example.

Also found a mistake in the solar federation faction which has had Mars as it's homeworld since it was created! 👍 
<!-- Please make sure you've read documentation on contributing -->

impaktor commented:
I'm not so sure Mars being the homeworld really was a mistake, initially, as I vaguely remember it being so in Frontier? Anyway, no objection from me there.

As for the capping tech level, it would be interesting to see the change as shown in #3505. Note: half the plots I made in that are now gone! Damn github! I had a histogram showing the tech levels in original frontier. Oh, well...

nozmajner commented:
I think Mars was intentional, since the ecosystem of Earth was wrecked, and now it is more of a reservation, to allow it to recover. [See 2328 - 2350.](https://pioneerwiki.com/wiki/Timeline_of_the_Pioneer_universe)

nozmajner commented:
I guess the Earth start was intended to provide a familiar place to start. But yeah, it would make sense to move it to Mars, so the player actually has to visit home.

fluffyfreak commented on data/libs/SpaceStation.lua:
```diff
@@ -27,6 +27,11 @@ function SpaceStation:Constructor()
+	-- cap the techlevel lower end based on the planets population
+	techLevel = math.max(techLevel, math.floor(self.path:GetSystemBody().parent.population * 0.5))
 	self:setprop("techLevel", techLevel)
 end
```
Population is in Billions so for Earth the value is ~7.4, Mars is only ~1.3 or something.
For this I just wanted high population worlds to get a base tech level boost.

impaktor replied:
What's the biggest population number? I.e. could this render "tech level 20" then, which is beyond what currently holds any meaning.

A player encountering something like that would (if they be true nerds) start documenting everything in the star port, to find what that extra tech level has given them, and would come up with zilch.

fluffyfreak replied:
Yes i guess that is an issue, I don't know of any, but that's because we don't have any good way of querying the game for that info reliably.

fluffyfreak replied:
Amended this so it caps the boost to 11

fluffyfreak commented:
Ok, so switch it back to Mars and change the starting location to be Mars as well?

fluffyfreak commented:
Suggestions for new Mars cities/orbitals welcome...

impaktor commented:
I'm not sure this is the time to move all/most bases from Earth to Mars, or? I assume it will be a save breaking change.

> Suggestions for new Mars cities/orbitals welcome...

So, not move, just create even more on Mars? Not sure I think that's the best way, as we're already spamming Sol like crazy with star ports. `more of same != more fun`.

fluffyfreak commented:
Not changing the cities on Mars for this one, just added the necessary strings and changed Mars to be the EASY start point, also the Solar Federation homeworld.

impaktor commented:
This also needs rebasing before merge, as it hurts my eyes with commits moving starting location around multiple times in the same PR.

(I can do that at merge time)

impaktor commented:
OK, two of the six commits were bug fixes / nerfs to the earlier commits of this same branch, so squashed them.

I don't quite know how this will affect tech levels, as in, what's the max/min pop level of a planet, and also, I'm not sure what the first commit alludes to, which "needless values" is it refering to, but fukk it, merged.
